docs/design/core/batch-utilities/Design.md

```diff
@@ -0,0 +1,345 @@
+# Github Design Document
+
```

Comment:
Can we add `**Design:** New Feature, **Status:** [Released](../../../README.md)`?
Example: https://github.com/aws/aws-sdk-java-v2/tree/master/docs/design/core/presigners